spec/unit/models/services/service_binding_spec.rb

```diff
@@ -75,13 +75,59 @@ module VCAP::CloudController
         end
       end
 
-      context 'when the service instance and app are in different spaces' do
+      describe 'service instance and app space matching' do
         let(:app) { AppModel.make }
-        let(:service_instance) { ManagedServiceInstance.make }
 
-        it 'is not valid' do
-          expect { ServiceBinding.make(service_instance: service_instance, app: app)
-          }.to raise_error(Sequel::ValidationFailed, /service_instance space_mismatch/)
+        context 'when the service instance and the app are in different spaces' do
+          let(:service_instance) { ManagedServiceInstance.make }
+          context 'when the service_instance_sharing feature flag is enabled' do
+            before do
+              VCAP::CloudController::FeatureFlag.create(name: :service_instance_sharing, enabled: true)
+            end
+
+            context 'when the service instance has not been shared into the app space' do
+              it 'is not valid' do
+                expect { ServiceBinding.make(service_instance: service_instance, app: app)
+                }.to raise_error(Sequel::ValidationFailed, /service_instance space_mismatch/)
+              end
+            end
+
+            context 'when the service instance has been shared into the app space' do
+              before do
+                service_instance.add_shared_space(app.space)
+              end
+
+              it 'it is valid' do
```

Comment:
apologies for nitpicking. stylistically can you remove duplicate `its` for consistency with rest of codebase?. --> it 'is valid' do ... end